readthedocs/config/config.py

```diff
@@ -889,11 +930,51 @@ def validate_submodules(self):
                 )
 
         with self.catch_validation_error('submodules.recursive'):
-            recursive = raw_submodules.get('recursive', False)
+            recursive = self.pop_config('submodules.recursive', False)
             submodules['recursive'] = validate_bool(recursive)
 
         return submodules
 
+    def validate_keys(self):
+        """
+        Checks that we don't have extra keys.
+
+        This should be called after all the validations are done.
```

Comment:
I think I got it now, but it's hard to read.

I suppose that when validating the keys we `pop` (remove) these keys from the `raw_config` and finally if there are some leftovers it means that we have invalid keys.

If this is the way it works, it would be good to clarify in the docstring.

Reply:
Yep, that is